### PR TITLE
feat(helpers): add stickers helpers

### PR DESCRIPTION
**Pull request by lts20050703**
Closes: #2461 
Docs: https://discord.com/developers/docs/resources/sticker#get-sticker

### CODE REVIEW

File: helpers/stickers/createGuildSticker.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
+ * Fires a Guild Stickers Update Gateway event.
+ * Every guilds has five free sticker slots by default, and each Boost level will grant access to more slots.
+ * Lottie stickers can only be uploaded on guilds that have either the `VERIFIED` and/or the `PARTNERED` guild feature.
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#get-guild-sticker}
+ */
+export async function createGuildSticker(
```
Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:20]: Needs a permission plugin for perm check

Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:20]: Needs a unit test

File: helpers/stickers/createGuildSticker.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * @param guildId The ID of the guild to get
+ * @return A {@link Sticker}
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
+ * Fires a Guild Stickers Update Gateway event.
+ * Every guilds has five free sticker slots by default, and each Boost level will grant access to more slots.
+ * Lottie stickers can only be uploaded on guilds that have either the `VERIFIED` and/or the `PARTNERED` guild feature.
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#get-guild-sticker}
```
Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:18]: Doesn't look like the right link

File: helpers/stickers/createGuildSticker.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    description: options.description,
+    tags: options.tags,
+    file: options.file,
+    reason: options.reason,
+  });
+  return bot.transformers.sticker(bot, result);
+}
+
+export interface CreateGuildStickerOptions extends WithReason {
+  /** Name of the sticker (2-30 characters) */
```
Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:36]: Needs validation plugin

File: helpers/stickers/createGuildSticker.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  name: string;
+  /** Description of the sticker (empty or 2-100 characters) */
+  description: string;
```
Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:39]: validation plugin

File: helpers/stickers/createGuildSticker.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  /** Autocomplete/suggestion tags for the sticker (max 200 characters) */
+  tags: string;
```
Comment (Skillz4Killz) [inline helpers/stickers/createGuildSticker.ts:41]: validation plugin

File: helpers/stickers/deleteGuildSticker.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Bot } from "../../bot.ts";
+import { Sticker } from "../../transformers/sticker.ts";
+
+/**
+ * Create a new sticker for the guild.
```
Comment (Skillz4Killz) [inline helpers/stickers/deleteGuildSticker.ts:5]: Delete*

File: helpers/stickers/deleteGuildSticker.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ * @param guildId The ID of the guild to get
+ * @return A {@link Sticker}
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
+ * Fires a Guild Stickers Update Gateway event.
+ * Every guilds has five free sticker slots by default, and each Boost level will grant access to more slots.
+ * Lottie stickers can only be uploaded on guilds that have either the `VERIFIED` and/or the `PARTNERED` guild feature.
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#get-guild-sticker}
```
Comment (Skillz4Killz) [inline helpers/stickers/deleteGuildSticker.ts:17]: Wrong link

File: helpers/stickers/deleteGuildSticker.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+ */
+export async function deleteGuildSticker(
```
Comment (Skillz4Killz) [inline helpers/stickers/deleteGuildSticker.ts:19]: unit test

File: helpers/stickers/deleteGuildSticker.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Create a new sticker for the guild.
+ *
+ * @param bot The bot instance to use to make the request.
+ * @param guildId The ID of the guild to get
+ * @return A {@link Sticker}
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
```
Comment (Skillz4Killz) [inline helpers/stickers/deleteGuildSticker.ts:12]: permission plugin

File: helpers/stickers/editGuildSticker.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Edit the given sticker.
+ *
+ * @param bot The bot instance to use to make the request.
+ * @param guildId The ID of the guild to get
+ * @return A {@link Sticker}
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
```
Comment (Skillz4Killz) [inline helpers/stickers/editGuildSticker.ts:14]: permisison plugin

File: helpers/stickers/editGuildSticker.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    name: options.name,
+    description: options.description,
+    tags: options.tags,
+    reason: options.reason,
+  });
+  return bot.transformers.sticker(bot, result);
+}
+
+export interface EditGuildStickerOptions extends WithReason {
+  /** Name of the sticker (2-30 characters) */
```
Comment (Skillz4Killz) [inline helpers/stickers/editGuildSticker.ts:35]: validation plugin

File: helpers/stickers/editGuildSticker.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  name?: string;
+  /** Description of the sticker (empty or 2-100 characters) */
+  description?: string | null;
```
Comment (Skillz4Killz) [inline helpers/stickers/editGuildSticker.ts:38]: validation

File: helpers/stickers/editGuildSticker.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  /** Autocomplete/suggestion tags for the sticker (max 200 characters) */
+  tags?: string;
```
Comment (Skillz4Killz) [inline helpers/stickers/editGuildSticker.ts:40]: validation

File: helpers/stickers/editGuildSticker.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+ * @param guildId The ID of the guild to get
+ * @return A {@link Sticker}
+ *
+ * @remarks
+ * Requires the `MANAGE_EMOJIS_AND_STICKERS` permission.
+ * Fires a Guild Stickers Update Gateway event.
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#modify-guild-sticker}
+ */
+export async function editGuildSticker(
```
Comment (Skillz4Killz) [inline helpers/stickers/editGuildSticker.ts:19]: unit test

File: helpers/stickers/getGuildSticker.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+ * @param bot The bot instance to use to make the request.
+ * @param guildId The ID of the guild to get
+ * @param stickerId The ID of the sticker to get
+ * @return A {@link Sticker}
+ *
+ * @remarks Includes the user field if the bot has the `MANAGE_EMOJIS_AND_STICKERS` permission.
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#get-guild-sticker}
+ */
+export async function getGuildSticker(bot: Bot, guildId: bigint, stickerId: bigint): Promise<Sticker> {
```
Comment (Skillz4Killz) [inline helpers/stickers/getGuildSticker.ts:15]: unit test

File: helpers/stickers/getGuildSticker.ts [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Returns a sticker object for the given guild and sticker IDs.
+ *
+ * @param bot The bot instance to use to make the request.
+ * @param guildId The ID of the guild to get
+ * @param stickerId The ID of the sticker to get
+ * @return A {@link Sticker}
+ *
+ * @remarks Includes the user field if the bot has the `MANAGE_EMOJIS_AND_STICKERS` permission.
```
Comment (Skillz4Killz) [inline helpers/stickers/getGuildSticker.ts:11]: unit test to check this

File: helpers/stickers/getGuildStickers.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
+import { Bot } from "../../bot.ts";
+import { Collection, Sticker } from "../../mod.ts";
+import { DiscordSticker } from "../../types/discord.ts";
+
+/**
+ * Returns an array of sticker objects for the given guild.
+ *
+ * @param bot The bot instance to use to make the request.
+ * @param stickerId The ID of the sticker to get
```
Comment (Skillz4Killz) [inline helpers/stickers/getGuildStickers.ts:9]: should be guild id not sticker id

File: helpers/stickers/getGuildStickers.ts [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+ * @returns A collection of {@link Sticker} objects assorted by sticker ID.
+ *
+ * @remarks Includes user fields if the bot has the `MANAGE_EMOJIS_AND_STICKERS` permission.
```
Comment (Skillz4Killz) [inline helpers/stickers/getGuildStickers.ts:12]: unit test

File: helpers/stickers/getGuildStickers.ts [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#list-guild-stickers}
+ */
+export async function getGuildStickers(bot: Bot, stickerId: bigint): Promise<Collection<bigint, Sticker>> {
```
Comment (Skillz4Killz) [inline helpers/stickers/getGuildStickers.ts:16]: unit test

Comment (Skillz4Killz) [inline helpers/stickers/getGuildStickers.ts:16]: guild id not sticker id

File: helpers/stickers/getNitroStickerPacks.ts [hunk 20]
```diff
@@ -3,7 +3,14 @@ import { StickerPack } from "../../transformers/sticker.ts";
 import { DiscordStickerPack } from "../../types/discord.ts";
```
Comment (Skillz4Killz) [inline helpers/stickers/getNitroStickerPacks.ts:3]: Move this file in a different pr, it makes it breaking.

File: helpers/stickers/getSticker.ts [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Returns a sticker object for the given sticker ID.
+ *
+ * @param bot The bot instance to use to make the request.
+ * @param stickerId The ID of the sticker to get
+ * @returns A {@link Sticker}
+ *
+ * @see {@link https://discord.com/developers/docs/resources/sticker#get-sticker}
+ */
+export async function getSticker(bot: Bot, stickerId: bigint): Promise<Sticker> {
```
Comment (Skillz4Killz) [inline helpers/stickers/getSticker.ts:14]: unit test

File: helpers/stickers/createGuildSticker.ts (hunk 2 above)
Comment (lts20050703) [reply helpers/stickers/createGuildSticker.ts:18]: oops copy paste my bad

File: plugins/permissions/src/stickers/createGuildSticker.ts [hunk 22]
```diff
@@ -0,0 +1,10 @@
+import { BotWithCache } from "../../deps.ts";
+import { requireBotGuildPermissions } from "../permissions.ts";
+
+export async function createGuildSticker(bot: BotWithCache) {
+  const createGuildSticker = bot.helpers.createGuildSticker;
+  bot.helpers.createGuildSticker = (guildId, options) => {
+    requireBotGuildPermissions(bot, guildId, ["MANAGE_EMOJIS"]);
```
Comment (Skillz4Killz) [inline plugins/permissions/src/stickers/createGuildSticker.ts:7]: is this the right perm? not MANAGE_EMOJIS_AND_STICKERS?

File: tests/stickers/createGuildSticker.test.ts [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+import { loadBot } from "../mod.ts";
+import { CACHED_COMMUNITY_GUILD_ID } from "../utils.ts";
+
+Deno.test("[stickers] Create guild sticker", async () => {
+  const bot = loadBot();
+  const sticker = await bot.helpers.createGuildSticker(CACHED_COMMUNITY_GUILD_ID, {
+    name: "sticker name",
+    description: "sticker description",
+    tags: "sticker tags",
+    file: { blob: new Blob(), name: "sticker file name" },
```
Comment (Skillz4Killz) [inline tests/stickers/createGuildSticker.test.ts:11]: Should this not fail if the filname doesnt have a valid file suffix like .png

File: plugins/permissions/src/stickers/createGuildSticker.ts (hunk 22 above)
Comment (lts20050703) [reply plugins/permissions/src/stickers/createGuildSticker.ts:7]: @Skillz4Killz #2473 haven't been merged yet

Comment (lts20050703) [reply plugins/permissions/src/stickers/createGuildSticker.ts:7]: Deno check is screaming because of BigString support, waiting for #2486 to merge before working on this

File: tests/stickers/createGuildSticker.test.ts (hunk 23 above)
Comment (lts20050703) [reply tests/stickers/createGuildSticker.test.ts:11]: @Skillz4Killz we do not have any emoji tests so there's no image/asset i can reuse to test
any ideas?
